File: content/app/mojo/mojo_init.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "content/app/mojo/mojo_init.h"

#include "base/logging.h"
#include "mojo/application_manager/application_manager.h"
#include "mojo/embedder/embedder.h"

namespace content {

void InitializeMojo() {
  mojo::embedder::Init();
  mojo::ApplicationManager::GetInstance();
}

}  // namespace content

File: mojo/public/dart/mojo_init.dart
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

library mojo_init;

import 'dart:async';

import 'core.dart' as core;
import 'dart-ext:src/mojo_dart_init';

void _init() native "MojoLibrary_Init";
void _mojoSystemThunksMake(Function fn) native "MojoSystemThunks_Make";

Future<Isolate> mojoInit() {
  _init();
  _mojoSystemThunksMake(core.mojoSystemThunksSet);
  return core.MojoHandleWatcher.Start();
}

void mojoShutdown() {
  core.MojoHandleWatcher.Stop();
}
